report/sections/Evaluation/Benchmark.tex

```diff
@@ -0,0 +1,202 @@
+The benchmark itself measures the time it takes for all clients to post the code submission and receive an HTTP response with status code 200.
+If just one of the submissions cannot be processed, an exception is thrown and the benchmark stops.
+It utilizes the \texttt{BuildClientTaskList} method seen in listing \ref{lst:baselineBench}.
+\begin{lstlisting}[language=CSharp, escapechar=~, caption={C\# code showing the benchmark implementation for the synchronous baseline}, label={lst:baselineBench}]
+// Attributes excluded
+public class RocketBenchmarks
+{
+    [Params(10, 20, 50, 100)] 
+    public int NumberOfRequests { get; set; }
+    [ParamsSource(nameof(CodeSubmissions))]
+    public CodeSubmission CodeSubmission { get; set; }
+    public static IEnumerable<CodeSubmission> CodeSubmissions => CodeLoader.Load();
+
+    [Benchmark]
+    public void PostAndWaitForResponseReceived()
+    {
+      IEnumerable<Task> clientActions = TaskBuilder.BuildClientTaskList<CodeRunnerClient>(NumberOfRequests,
+        client => client.Post(CodeSubmission)
+        .Result.EnsureSuccessStatusCode());
+        
+      Task.WhenAll(clientActions).Wait();
+    }
+}
+\end{lstlisting}
+
+\subsubsection{Queue system benchmark implementation}
+The benchmark for the queue system seen in listing \ref{lst:queueSystemBench} is similar to the one in listing \ref{lst:baselineBench}, as it utilizes the method from listing \ref{lst:TaskBuilder} and the parameterization attributes from BenchmarkDotNet.
+However, the internal implementation of the client varies, as it needs to poll for the results of the Test Runner after making an initial request.
+Therefore, an additional parameter for the benchmark is introduced to examine how often a client should try to poll for a result.
+Best case poll time needs to be established to compare it with the baseline component. 
+However, due to the nature of online remote communication, it may be necessary to establish which polling times to avoid instead of using the best average response times.
+This is because the benchmarks do not consider the physical distance between the client and the server.
+\begin{lstlisting}[language=CSharp, escapechar=~, caption={C\# code showing the benchmark implementation for the asynchronous backend}, label={lst:queueSystemBench}]
+// Attributes excluded
+public class TicketedCodeRunnerBenchmark
+{
+    [Params(0.5, 1.0, 2.0, 3.0)] 
+    public double PollTime { get; set; }
+    [Params(10, 20, 50, 100)] 
+    public int NumberOfRequests { get; set; }
+    [ParamsSource(nameof(CodeSubmissions))] 
+    public CodeSubmission CodeSubmission { get; set; }
+    public static IEnumerable<CodeSubmission> CodeSubmissions => CodeLoader.Load();
+
+    [Benchmark]
+    public void PostAndWaitForAllResultsFetched()
+    {
+      TimeSpan timeBetweenPulls = TimeSpan.FromSeconds(PollTime);
```

Comment:
`timeBetweenPulls` needs to be `timeBetweenPolls`. You need to correct usage of pull to poll in both the C# code and the code snippets in the report.